### ISSUE TITLE
Job status incorrectly shows as "running" in TUI/GUI on Cylc 8.2 for certain XTrigger use cases

### ISSUE THREAD

**Issue by mo-jareddrayton** [cylc/cylc-flow#5645]
### Description

On `cylc 8.2` a task's job status remains on `running` instead of `succeeded` in the `gui` and `tui` for cases when there are no further tasks dependent on that task but there is still an external trigger to be satisfied. This is in contrast to `cylc 8.1.4` where the task's job status is correctly updated promptly in both the `gui` and `tui` once the job has succeeded.

I also notice that
- Cylc Review correctly show the job as `succeeded`.
- The manual `poll` option in the `gui` and `tui` will correctly update the job status.

### Reproducible Example

Run the following workflow with `8.2` and `8.1.4` respectively to observe the difference in behaviour.

`cylc vip . --workflow-name job_status_8.2`

```
[scheduling]
    [[xtriggers]]
        initialise_work_succeeded = workflow_state(job_status_8.2/runN, 'initialise_work', '1'):PT60S
    [[graph]]
        R1 = initialise_work => do_work
        R1 = @initialise_work_succeeded => complete_work
[runtime]
    [[root]]
        script = sleep 5
    [[initialise_work]]
    [[do_work]]
    [[complete_work]]
```

### Expected Behaviour

I would expect the task status to update in the same way it does for `cylc 8.1.4`.

**Comment by wxtim** [comment]
Replicated this with the example given.

**Comment by wxtim** [comment]
This change was introduced in https://github.com/cylc/cylc-flow/pull/5231

**Comment by hjoliver** [comment]
Hmm, this rings a bell. I think I saw it a while back, and wrote this:
> https://github.com/cylc/cylc-flow/pull/5231#issuecomment-1637791144

Unfortunately I didn't explain the reason, then went on leave on forgot about it. 😬

**Comment by oliver-sanders** [comment]
Closed by #5650 

We'll try and get this released promptly.

Thanks for the bug report and the great example @mo-jareddrayton
